MekHQ/src/mekhq/campaign/autoresolve/acar/manager/PhaseEndManager.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2024 - The MegaMek Team. All Rights Reserved.
+ *
+ *  This file is part of MekHQ.
+ *
+ *  MekHQ is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  MekHQ is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package mekhq.campaign.autoresolve.acar.manager;
+
+import megamek.common.enums.GamePhase;
+import mekhq.campaign.autoresolve.acar.SimulationManager;
+import mekhq.campaign.autoresolve.acar.report.PublicReportEntry;
+
+public record PhaseEndManager(SimulationManager simulationManager) implements SimulationManagerHelper {
+
+    public void managePhase() {
+        switch (simulationManager.getGame().getPhase()) {
```

Comment:
If you used an enhanced switch here you wouldn't need to include `break` after each case, removing a lot of risk from writing switches. Generally, post 50.0, we should only be using normal switches in cases where we want specific fall through behavior - as enhanced switches don't allow for fall through.

Reply:
I am banking on fallthrough behavior in this switch statement

```java
switch (game().getPhase()) {
            case DEPLOYMENT:
            case SBF_DETECTION:
            case MOVEMENT:
            case FIRING:
                resetEntityPhase();
                simulationManager.initiativeHelper.determineTurnOrder(game().getPhase());
                case INITIATIVE:
                clearActions();
                break;
                
                
            case END:
            case VICTORY:
            default:
                clearReports();
                clearActions();
                break;